### PR TITLE
pack.py: add 'cwlVersion' in pack dict, if only single item exists in $graph

### PR DESCRIPTION
**Pull request by manu-chroma**
Closes https://github.com/common-workflow-language/cwltool/issues/482

### CODE REVIEW

**Comment by manu-chroma** [comment]
``` bash
(venv) [venv] ~/github/test_cwl_runner/workflows/workflows/hello · (master±) $ 
cwltool --pack hello-param.cwl params.yaml
/home/manu/github/cwltool/venv/bin/cwltool 1.0.20170725170438
Resolved 'hello-param.cwl' to 'file:///home/manu/github/test_cwl_runner/workflows/workflows/hello/hello-param.cwl'
{
    "cwlVersion": "v1.0", 
    "$graph": [
        {
            "class": "Workflow", 
            "label": "Hello World", 
            "doc": "Outputs a message using echo", 
            "inputs": [
                {
                    "type": "string", 
                    "id": "#main/usermessage"
                }
            ], 
            "outputs": [
                {
                    "outputSource": "#main/step0/response", 
                    "type": "File", 
                    "id": "#main/response"
                }
            ], 
            "steps": [
                {
                    "run": {
                        "class": "CommandLineTool", 
                        "inputs": [
                            {
                                "type": "string", 
                                "doc": "The message to print", 
                                "default": "Hello World", 
                                "inputBinding": {
                                    "position": 1
                                }, 
                                "id": "#main/step0/b272aa88-48b7-4bed-a98a-061350c945d1/message"
                            }
                        ], 
                        "baseCommand": "echo", 
                        "arguments": [
                            "-n", 
                            "-e"
                        ], 
                        "stdout": "response.txt", 
                        "outputs": [
                            {
                                "type": "File", 
                                "id": "#main/step0/b272aa88-48b7-4bed-a98a-061350c945d1/response", 
                                "outputBinding": {
                                    "glob": "response.txt"
                                }
                            }
                        ], 
                        "id": "#main/step0/b272aa88-48b7-4bed-a98a-061350c945d1"
                    }, 
                    "in": [
                        {
                            "source": "#main/usermessage", 
                            "id": "#main/step0/message"
                        }
                    ], 
                    "out": [
                        "#main/step0/response"
                    ], 
                    "id": "#main/step0"
                }
            ], 
            "id": "#main"
        }
    ]
}⏎ 
```

**Comment by manu-chroma** [comment]
@FarahZKhan Hey, can you test this branch on some other workflows with ``--pack`` to see if it fixes the issue?
Thanks. :)

File: cwltool/main.py [hunk 1]
```diff
@@ -655,10 +655,8 @@ def loadref(b, u):
 def print_pack(document_loader, processobj, uri, metadata):
     # type: (Loader, Union[Dict[Text, Any], List[Dict[Text, Any]]], Text, Dict[Text, Any]) -> str
     packed = pack(document_loader, processobj, uri, metadata)
-    if len(packed["$graph"]) > 1:
-        return json.dumps(packed, indent=4)
-    else:
-        return json.dumps(packed["$graph"][0], indent=4)
+
+    return json.dumps(packed, indent=4)
```
Comment (manu-chroma) [inline cwltool/main.py]: I don't think this conditional statement had any significance at this place. 
@mr-c Something I might be missing out?

Comment (mr-c) [inline cwltool/main.py]: This is needed -- if there is only a single item in the `graph` then it is output on its own, without the `$graph` wrapped around it

**Comment by manu-chroma** [comment]
Is it a problem if we've "cwlVersion" printed at the bottom of the ``--pack`` output?

``` bash
cwltool --pack hello-param.cwl params.yaml
/home/manu/github/cwltool/venv/bin/cwltool 1.0.20170725170438
Resolved 'hello-param.cwl' to 'file:///home/manu/github/test_cwl_runner/workflows/workflows/hello/hello-param.cwl'
{
    "class": "Workflow", 
    "label": "Hello World", 
    "doc": "Outputs a message using echo", 
    "inputs": [
        {
            "type": "string", 
            "id": "#main/usermessage"
        }
    ], 
    "outputs": [
        {
            "outputSource": "#main/step0/response", 
            "type": "File", 
            "id": "#main/response"
        }
    ], 
    "steps": [
        {
            "run": {
                "class": "CommandLineTool", 
                "inputs": [
                    {
                        "type": "string", 
                        "doc": "The message to print", 
                        "default": "Hello World", 
                        "inputBinding": {
                            "position": 1
                        }, 
                        "id": "#main/step0/caddf27e-7a05-4f06-b2b0-a011b3b27718/message"
                    }
                ], 
                "baseCommand": "echo", 
                "arguments": [
                    "-n", 
                    "-e"
                ], 
                "stdout": "response.txt", 
                "outputs": [
                    {
                        "type": "File", 
                        "id": "#main/step0/caddf27e-7a05-4f06-b2b0-a011b3b27718/response", 
                        "outputBinding": {
                            "glob": "response.txt"
                        }
                    }
                ], 
                "id": "#main/step0/caddf27e-7a05-4f06-b2b0-a011b3b27718"
            }, 
            "in": [
                {
                    "source": "#main/usermessage", 
                    "id": "#main/step0/message"
                }
            ], 
            "out": [
                "#main/step0/response"
            ], 
            "id": "#main/step0"
        }
    ], 
    "id": "#main", 
    "cwlVersion": "v1.0"
}⏎  
```

File: cwltool/main.py (hunk 1 above)
Comment (manu-chroma) [inline cwltool/main.py]: Thanks for clarifying this one :D

**Comment by mr-c** [comment]
@manu_chroma: the order doesn't matter, not a problem

File: tests/wf/hello-workflow.cwl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cwlVersion: v1.0
```
Comment (manu-chroma) [inline tests/wf/hello-workflow.cwl]: @mr-c does this qualify as both? a single tool, simple workflow.

Comment (mr-c) [inline tests/wf/hello-workflow.cwl]: The point is to traverse both branches of https://github.com/common-workflow-language/cwltool/blob/23ad7342896e0a46d7e25ee479f764ba60c61983/cwltool/main.py#L658

**Comment by manu-chroma** [comment]
I have this type of [pack output](https://github.com/common-workflow-language/cwltool/pull/497#issuecomment-317962503) from simple workflow which I am comparing against the expected. If I am correct, I should delete the ``"id"`` key before doing assertion in the test. Right?

**Comment by manu-chroma** [comment]
@mr-c Can you review this one? I've added appropriate tests now

File: tests/test_pack.py [hunk 3]
```diff
@@ -33,3 +33,29 @@ def test_pack(self):
...
+        # Since diff is longer than 3174 characters
+        self.maxDiff = None
+
+        # Testing single tool workflow
+        document_loader, workflowobj, uri = fetch_document(
+            get_data("tests/wf/hello_single_tool.cwl"))
+        document_loader, avsc_names, processobj, metadata, uri = validate_document(
+            document_loader, workflowobj, uri)
+        # generate pack output dict
+        packed = cwltool.pack.pack(document_loader, processobj, uri, metadata)
```
Comment (mr-c) [inline tests/test_pack.py]: it would be better to test the output of `cwltool.main.print_pack()`

Comment (manu-chroma) [inline tests/test_pack.py]: Done :+1:

**Comment by manu-chroma** [comment]
@mr-c Does it look mergeable now?

**Comment by mr-c** [comment]
Thanks!

**Pull request merged**